downloads/views.py

```diff
@@ -48,20 +69,60 @@ def get_presigned_url(
     )
 
 
-def stat_object(job_id: str, task_id: str, node_id: str, timestamp_and_filename: str):
+def stat_osn_object(item: Item):
     client = get_minio_client()
-    object_name = get_object_name(job_id, task_id, node_id, timestamp_and_filename)
+    object_name = get_osn_object_name(item)
     return client.stat_object(
         bucket_name=settings.S3_BUCKET_NAME,
         object_name=object_name,
     )
 
 
-def get_timestamp_for_timestamp_and_filename_string(s: str) -> datetime:
-    s = s.split("-", maxsplit=1)[0]
-    nanos = float(s)
-    timestamp = nanos / 1e9
-    return datetime.fromtimestamp(timestamp, tz=timezone.utc)
+def object_exists_in_osn(item: Item):
+    client = get_minio_client()
+    object_name = get_osn_object_name(item)
+
+    try:
+        client.stat_object(
+            bucket_name=settings.S3_BUCKET_NAME,
+            object_name=object_name,
+        )
+        return True
+    except S3Error:
+        return False
+
+
+def get_pelican_path(item: Item):
+    return f"/node-data/{item.job_id}/{item.task_id}/{item.node_id}/{item.timestamp_and_filename}"
+
+
+def get_pelican_authz_url(item: Item):
+    path = get_pelican_path(item)
+
+    key = Path(settings.PELICAN_KEY).read_bytes()
+
+    token = SciToken(
+        key=key,
+        algorithm=settings.PELICAN_ALGORITHM,
```

Comment:
Instead of hardcoding this in the settings, I used our public key configuration endpoint to retrieve it. Of course, adding it here directly isn't logical because it would require an additional call every time an object is retrieved. However, we can add it one layer up, so it doesn't need to call the public key configuration endpoint each time an object is downloaded. It's not important now since I'm sure we won't be changing our algorithm much but something to consider.

Reply:
Thank you. This is an interesting point.. I'll make a note to revisit decide the best way to handle this.

You're totally right that renewing this would be _very_ infrequent... likely never unless something was compromised. Additionally, if we renewed that, we'd have to update this service anyway since it uses the corresponding issuer key.

Maybe for a first pass, I'll keep it this way and then think through the best way to fetch. (Ex. fetch once at start, fetch each call, fetch and cache when successful).

Reply:
FYI I think the addition itself will be easy - I just want to make sure we don't accidentally do something like add fetch once at start but then because something on Github Pages is down, the auth service doesn't start because it fails to fetch the config.